### DISCUSSION TITLE
Support for user-defined procedural collision?

### DISCUSSION THREAD

**Discussion by UltraEngine** [jrouwe/JoltPhysics#705] [Q&A]
I was wondering if this library supports user-defined procedural collision shapes? This would work by defining a callback that accepts a bounding box and returns a list of triangles or polygons that can be collided within that bounding box. This is useful for advanced terrain systems and procedural vegetation.

**Comment by jrouwe** [comment]
Yes it does, please see the `Creating Custom Shapes` section [here](https://jrouwe.github.io/JoltPhysics/).
